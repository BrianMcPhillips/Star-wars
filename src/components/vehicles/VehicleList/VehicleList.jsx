import React from 'react';
import PropTypes from 'prop-types';
import Vehicle from '../Vehicle/Vehicle';

const VehicleList = ({ data }) => {
  const vehicleStuff = data.map((vehicle, i) => 
    <li key={vehicle + i}>
      <Vehicle {...vehicle} />
    </li>
  );

  return (
    <ul>
      {vehicleStuff}
    </ul>
  );
};

VehicleList.propTypes = {
  data: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    model: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
  })).isRequired
};

export default VehicleList;
