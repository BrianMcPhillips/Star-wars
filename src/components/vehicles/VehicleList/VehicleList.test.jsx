import React from 'react';
import { render, cleanup } from '@testing-library/react';
import VehicleList from './VehicleList';

describe('VehicleList component', () => {
  afterEach(() => cleanup());
  it('renders VehicleList', () => {
    const { asFragment } = render(
      <VehicleList data= {[
        {
          name: 'Speeder',
          model: 'Imperial I-class Speeder',
          url: 'https://swapi.dev/api/vehicle/3/'
        },
        {
          name: 'Sentinel-class landing craft',
          model: 'Sentinel-class landing craft',
          url: 'https://swapi.dev/api/vehicle/5/'
        }
      ]}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
