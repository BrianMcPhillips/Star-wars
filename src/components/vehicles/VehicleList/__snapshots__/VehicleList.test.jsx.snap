// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VehicleList component renders VehicleList 1`] = `
<DocumentFragment>
  <ul>
    <li>
      <h1>
        Speeder
      </h1>
      <h4>
        Imperial I-class Speeder
      </h4>
    </li>
    <li>
      <h1>
        Sentinel-class landing craft
      </h1>
      <h4>
        Sentinel-class landing craft
      </h4>
    </li>
  </ul>
</DocumentFragment>
`;
